File: flutter_seller/lib/models/combo.dart
class ComboItem {
  final int productId;
  final String productName;
  final String productImage;
  final double productPrice;
  final int quantity;

  const ComboItem({
    required this.productId,
    required this.productName,
    required this.productImage,
    required this.productPrice,
    required this.quantity,
  });

  factory ComboItem.fromJson(Map<String, dynamic> json) => ComboItem(
        productId: json['product_id'] as int,
        productName: json['product_name'] as String? ?? '',
        productImage: json['product_image'] as String? ?? '',
        productPrice: (json['product_price'] as num?)?.toDouble() ?? 0.0,
        quantity: json['quantity'] as int? ?? 1,
      );
}

class Combo {
  final int id;
  final String name;
  final String description;
  final double price;
  final String imageUrl;
  final bool isActive;
  final int? branchId;
  final List<ComboItem> items;

  const Combo({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.imageUrl,
    required this.isActive,
    this.branchId,
    required this.items,
  });

  factory Combo.fromJson(Map<String, dynamic> json) => Combo(
        id: json['id'] as int,
        name: json['name'] as String,
        description: json['description'] as String? ?? '',
        price: (json['price'] as num).toDouble(),
        imageUrl: json['image_url'] as String? ?? '',
        isActive: json['is_active'] as bool? ?? true,
        branchId: json['branch_id'] as int?,
        items: ((json['items'] as List<dynamic>?) ?? [])
            .map((e) => ComboItem.fromJson(e as Map<String, dynamic>))
            .toList(),
      );

  String get itemsSummary =>
      items.map((i) => '${i.productName} x${i.quantity}').join(' + ');
}

File: app/models/combo.py
from ..database.db import db
from ..utils import now_mx


class Combo(db.Model):
    __tablename__ = "combos"
    id          = db.Column(db.Integer, primary_key=True)
    name        = db.Column(db.String(120), nullable=False)
    description = db.Column(db.String(300), default="")
    price       = db.Column(db.Float, nullable=False)
    image_url   = db.Column(db.String(300), default="")
    is_active   = db.Column(db.Boolean, default=True)
    branch_id   = db.Column(db.Integer, db.ForeignKey("branches.id"), nullable=True)
    created_at  = db.Column(db.DateTime, default=now_mx)

    items = db.relationship(
        "ComboItem", backref="combo", lazy=True, cascade="all, delete-orphan"
    )

    def to_dict(self):
        return {
            "id": self.id,
            "name": self.name,
            "description": self.description or "",
            "price": self.price,
            "image_url": self.image_url or "",
            "is_active": self.is_active,
            "branch_id": self.branch_id,
            "created_at": self.created_at.isoformat(),
            "items": [ci.to_dict() for ci in self.items],
        }


class ComboItem(db.Model):
    __tablename__ = "combo_items"
    id         = db.Column(db.Integer, primary_key=True)
    combo_id   = db.Column(db.Integer, db.ForeignKey("combos.id"), nullable=False)
    product_id = db.Column(db.Integer, db.ForeignKey("products.id"), nullable=False)
    quantity   = db.Column(db.Integer, default=1, nullable=False)

    product = db.relationship("Product", lazy=True)

    def to_dict(self):
        p = self.product
        return {
            "id": self.id,
            "product_id": self.product_id,
            "product_name": p.name if p else "",
            "product_image": p.image_url if p else "",
            "product_price": p.price if p else 0.0,
            "quantity": self.quantity,
        }
